### ISSUE TITLE
improvement of compatibility between copyleft licences?

### ISSUE THREAD

**Issue by owlshrimp** [copyleft-next/copyleft-next#55]
One of the things that's bugged me about the GPL is that it's generally incompatible all other copyleft licences (eg, the one used by OpenZFS). The spirit of both licences is the same, but due to particular details in their conditions, code can't be taken from one to the other (ie. OpenZFS can't become part of the GPLv2 linux kernel).

In an ideal world, maybe one could specify a list of requirements that a "copyleft licence" should meet, and to claim compatibility with that? Then any compliant copyleft licences could shuttle code among themselves.

I fully expect that this would completely break #38 (GPL compatibility) however, and possibly isn't even practical.

For a slightly more relaxed requirements-based "compatibility" copyleft licence (or class of licences) I'm guessing:
A) You can't accept GPL code, because it might be incompatible with other code already incorporated
B) You can't export to GPL code, because you might have GPL incompatible code incorporated
C) This might not even work at all, because the GPL is probably not the only copyleft licence with terms incompatible with other copyleft licences (or possibly even various "permissive licences"). There's probably lots of licences that can't be combined under one licence no matter what.

It seems the GPL was designed with the expectation of having a monopoly on copyleft (and limits what a GPL-compatible licence can do). I would like to explore the idea of better cross-copyleft-licence compatibility, but it seems like the kind of discussion that should have been had 40 years ago when all the currently-existing licences were being drafted. Would it even be worthwhile to include latent compatibility on the off chance it helps in the future? (or is this essentially just fragmenting the licenced code by requiring careful tracking of each file's history?)

**Comment by ghost** [comment]
But technically both the CDDL and GPL are FSF approved licenses so it wouldn't be an issue if the FSF drafted GPLv4 to allow CDDL and other FSF approved licenses into the mix

**Comment by ghost** [comment]
Or a modified GPLv2

**Comment by owlshrimp** [comment]
Non-GPL Free-Software copyleft licences are recognized by the FSF but they aren't recommended. I wouldn't expect the FSF to extend support to them as it would weaken their monopoly position as controllers of the GPL, in particular through the common "or any later version" statement.

**Comment by owlshrimp** [comment]
Kinda makes me wonder if there's a place out there for some kind of copyleft-macro licence. Essentially the licence equivalent of a preprocessor macro that says "you can treat this code as being under any one of this list of copyleft licences."

ie.
`// SPDX-License-Identifier: GPL-2.0 OR GPL-3.0 OR CDDL-1.0 OR CDDL-1.1` etc
